WordPressAuthenticator/Analytics/SignInTracker.swift

```diff
@@ -0,0 +1,345 @@
+import Foundation
+
+/// Implements the analytics tracking logic for our sign in flow.
+///
+public class SignInTracker {
+    
+    /// The method used for analytics tracking.  Useful for overriding in automated tests.
+    ///
+    typealias TrackerMethod = (_ event: AnalyticsEvent) -> ()
+
+    enum EventType: String {
+        case step = "unified_login_step"
+        case interaction = "unified_login_interaction"
+        case failure = "unified_login_failure"
+    }
+    
+    enum Property: String {
+        case failure
+        case flow
+        case click
+        case source
+        case step
+    }
+    
+    enum Source: String {
+        case `default`
+        case jetpack
+        case share
+        case deeplink
+        case reauthentication
+        
+        /// Starts when the used adds a site from the site picker
+        ///
+        case selfHosted
+    }
+    
+    enum Flow: String {
+        /// The initial flow before we decide whether the user is logging in or signing up
+        case wpCom = "wordpress_com"
+        
+        /// The flow that starts when the user starts the Google login
+        ///
+        case googleLogin = "google_login"
+        
+        case googleSignup = "google_signup"
+        
+        /// Flow for Sign in with Apple
+        ///
+        case apple = "siwa_login"
```

Comment:
We'll also need `siwa_signup`. 

Pinging @ScoutHarris to confirm / deny that ☝️ is true.

Reply:
Agreed. We can determine if it's login or signup in `AppleAuthenticator` (as we do currently), so we should add `siwa_signup`. Thanks @mindgraffiti !

Reply:
Signup for Apple added.